filament/src/Renderer.cpp

```diff
@@ -880,11 +882,16 @@ FrameGraphId<FrameGraphTexture> FRenderer::colorPass(FrameGraph& fg, const char*
 
                 // We set a "read" constraint on these attachments here because we need to preserve them
                 // when the color pass happens in several passes (e.g. with SSR)
+                auto depthAttachmentUsage = FrameGraphTexture::Usage::DEPTH_ATTACHMENT;
+                if (MTLPixelFormatDepth32Float_Stencil8 == config.depthFormat || MTLPixelFormatDepth24Unorm_Stencil8 == config.depthFormat) {
+                    depthAttachmentUsage |= FrameGraphTexture::Usage::STENCIL_ATTACHMENT;
+                }
+
                 data.color = builder.read(data.color, FrameGraphTexture::Usage::COLOR_ATTACHMENT);
-                data.depth = builder.read(data.depth, FrameGraphTexture::Usage::DEPTH_ATTACHMENT);
+                data.depth = builder.read(data.depth, depthAttachmentUsage | clearStencilFlags);
 
                 data.color = builder.write(data.color, FrameGraphTexture::Usage::COLOR_ATTACHMENT);
-                data.depth = builder.write(data.depth, FrameGraphTexture::Usage::DEPTH_ATTACHMENT);
+                data.depth = builder.write(data.depth, depthAttachmentUsage | clearStencilFlags);
```

Comment:
I suppose the two `... |  clearStencilFlags` parts were intended to be placed a few lines lower, where `.clearFlags` properties are set. The parameter here should be simply `depthAttachmentUsage`. (The two variables are incompatible types, so `operator|` is failing).

Reply:
fixed in https://github.com/shapr3d/filament/pull/58/commits/96d6ff120e6cb7e6eb5c5f37789eb0600043d643